plugins/SpectrumAnalyzer/SaSpectrumView.h

```diff
@@ -115,12 +115,19 @@ private slots:
 	float freqToXPixel(float frequency, unsigned int width);
 	float ampToYPixel(float amplitude, unsigned int height);
 
+	bool isRenderLogarithmicAmpTics() const;
+	std::vector<std::pair<float, std::string>> const & getAmpTicsToRender() const;
+
+	bool isRenderLogarithmicFrequencyTics() const;
+	std::vector<std::pair<int, std::string>> const & getFrequencyTicsToRender() const;
```

Comment:
Since you use `std::pair<int, std::string>` many times, I suggest you make a typedef of it (something like `FrequencyAndText`), and the same for `std::pair<float, std::string>`. 

Also, is there a reason for using std::string instead of QString?

Reply:
Using `std::string` would be mainly my doing, I used it everywhere in the code. I never worked with Qt before writing the analyzer and I didn't see any reason to use some weird non-standard data types when the `std` library works just fine. The same goes for containers (e.g. std::vector and not QVector). Maybe there is some reason why Qt had to reinvent everything (to the point you could almost call the language Qt++), but to this day the reason eludes me. :)

Reply:
In most cases it has the same interface (and more), In this case you have to manually convert it using `QString::fromStdString` which is just redundant.